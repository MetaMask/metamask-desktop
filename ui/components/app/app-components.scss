/** Please import your files in alphabetical order **/
@import 'account-list-item/index';
@import 'account-menu/index';
@import 'add-network/index';
@import 'advanced-gas-inputs/index';
@import 'app-loading-spinner/index';
@import 'import-token-link/index';
@import 'advanced-gas-controls/index';
@import 'alerts/alerts';
@import 'app-header/index';
@import 'asset-list-item/asset-list-item';
@import 'cancel-speedup-popover/index';
@import 'confirm-page-container/index';
@import 'confirm-page-container/enableEIP1559V2-notice';
@import 'collectibles-items/index';
@import 'collectibles-tab/index';
@import 'collectible-details/index';
@import 'collectible-default-image/index';
@import 'collectible-options/index';
@import 'collectibles-detection-notice/index';
@import 'connected-accounts-list/index';
@import 'connected-accounts-permissions/index';
@import 'connected-sites-list/index';
@import 'connected-status-indicator/index';
@import 'create-new-vault/create-new-vault.scss';
@import 'custom-spending-cap/index';
@import 'deposit-popover/index';
@import 'edit-gas-display/index';
@import 'edit-gas-display-education/index';
@import 'edit-gas-fee-button/index';
@import 'edit-gas-fee-popover/index';
@import 'edit-gas-fee-popover/edit-gas-item/index';
@import 'edit-gas-fee-popover/network-statistics/index';
@import 'edit-gas-fee-popover/network-statistics/status-slider/index';
@import 'edit-gas-fee-popover/edit-gas-tooltip/index';
@import 'flask/experimental-area/index';
@import 'flask/snaps-authorship-pill/index';
@import 'flask/snap-content-footer/index';
@import 'flask/snap-install-warning/index';
@import 'flask/snap-remove-warning/index';
@import 'flask/snap-settings-card/index';
@import 'flask/update-snap-permission-list/index';
@import 'gas-details-item/index';
@import 'gas-details-item/gas-details-item-title/index';
@import 'gas-timing/index';
@import 'hold-to-reveal-button/index';
@import 'home-notification/index';
@import 'info-box/index';
@import 'menu-bar/index';
@import 'modal/index';
@import 'modals/index';
@import 'multilayer-fee-message/index';
@import 'multiple-notifications/index';
@import 'network-display/index';
@import 'permission-page-container/index';
@import 'permissions-connect-footer/index';
@import 'permissions-connect-header/index';
@import 'permissions-connect-permission-list/index';
@import 'recovery-phrase-reminder/index';
@import 'step-progress-bar/index.scss';
@import 'selected-account/index';
@import 'signature-request/index';
@import 'signature-request-siwe/index';
@import 'signature-request-original/index';
@import 'srp-input/srp-input';
@import 'tab-bar/index';
@import 'token-cell/token-cell';
@import 'token-list-display/token-list-display';
@import 'transaction-activity-log/index';
@import 'transaction-activity-log/transaction-activity-log-icon/index';
@import 'transaction-breakdown/index';
@import 'transaction-detail/index';
@import 'transaction-detail-item/index';
@import 'transaction-icon/transaction-icon';
@import 'transaction-list-item-details/index';
@import 'transaction-list-item/index';
@import 'transaction-list/index';
@import 'transaction-status/index';
@import 'transaction-total-banner/index';
@import 'wallet-overview/index';
@import 'whats-new-popup/index';
@import 'loading-network-screen/index';
@import 'transaction-decoding/index';
@import 'advanced-gas-fee-popover/index';
@import 'advanced-gas-fee-popover/advanced-gas-fee-gas-limit/index';
@import 'advanced-gas-fee-popover/advanced-gas-fee-inputs/index';
@import 'advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/index';
@import 'advanced-gas-fee-popover/advanced-gas-fee-inputs/priority-fee-input/index';
@import 'advanced-gas-fee-popover/advanced-gas-fee-input-subtext/index';
@import 'advanced-gas-fee-popover/advanced-gas-fee-defaults/index';
@import 'currency-input/index';
@import 'asset-list/detetcted-tokens-link/index';
@import 'detected-token/detected-token-address/index';
@import 'detected-token/detected-token-aggregators/index';
@import 'detected-token/detected-token-values/index';
@import 'detected-token/detected-token-details/index';
@import 'detected-token/detected-token-ignored-popover/index';
@import 'detected-token/detected-token-selection-popover/index';
@import 'network-account-balance-header/index';
@import 'approve-content-card/index';
