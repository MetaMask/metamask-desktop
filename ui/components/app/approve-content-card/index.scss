.approve-content-card-container {
  &__card,
  &__card--no-border {
    border-bottom: 1px solid var(--color-border-default);
    position: relative;
    padding-inline-start: 24px;
    padding-inline-end: 24px;
  }

  &__card--no-border {
    border-bottom: none;
  }

  &__card-header {
    position: relative;

    &__symbol {
      width: auto;
    }

    &__symbol--aligned {
      width: 100%;
    }

    &__title {
      width: 100%;
    }

    &__title--aligned {
      margin-inline-start: 27px;
      position: absolute;
      width: auto;
    }
  }

  &__card-content--aligned {
    margin-inline-start: 42px;
  }

  &__transaction-details-extra-content {
    width: 100%;
  }

  &__data {
    width: 100%;

    &__data-block {
      overflow-wrap: break-word;
    }
  }
}
