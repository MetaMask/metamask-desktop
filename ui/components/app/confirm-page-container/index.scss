@import 'confirm-page-container-content/index';
@import 'confirm-page-container-header/index';
@import 'confirm-detail-row/index';
@import 'confirm-page-container-navigation/index';

.confirm-page-container {
  &__dialog {
    margin: 12px 16px;
  }
}

.page-container {
  &__content-component-wrapper {
    height: 100%;
  }

  &__link {
    background: transparent;
    border: 0 transparent;
    display: inline;
    padding: 0;
    font-size: $font-size-h7;
  }
}
