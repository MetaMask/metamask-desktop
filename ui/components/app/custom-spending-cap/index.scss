.custom-spending-cap {
  &__input-value-and-token-name {
    display: contents;
  }

  &__max-button {
    position: relative;
  }

  &__input {
    width: 100%;

    &--button {
      background: none;
      color: var(--color-primary-default);
    }

    &--max-button {
      color: var(--color-text-alternative);
      background: none;
      position: absolute;
      margin-top: 55px;
      margin-inline-start: -75px;
    }
  }

  #custom-spending-cap-input-value {
    color: var(--color-error-default);
    padding-inline-end: 60px;
  }
}
