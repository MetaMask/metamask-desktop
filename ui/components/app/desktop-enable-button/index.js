export { default } from './desktop-enable-button.component';
