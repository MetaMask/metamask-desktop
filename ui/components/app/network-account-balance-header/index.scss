.network-account-balance-header {
  border-top: 1px solid var(--color-border-muted);
  border-bottom: 1px solid var(--color-border-muted);

  &__network-account {
    &__ident-icon-ethereum {
      margin-inline-start: -10px;
      margin-top: -20px;
    }
  }
}
