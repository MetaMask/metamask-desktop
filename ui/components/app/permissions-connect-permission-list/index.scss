.permissions-connect-permission-list {
  width: 100%;

  .permission {
    @include H6;

    width: 100%;
    padding-bottom: 16px;
    border-bottom: 1px solid var(--color-border-default);
    display: flex;
    flex-direction: row;
    align-items: center;
    color: var(--color-text-default);

    i {
      display: block;
      margin: 16px;
      min-width: 16px;
      min-height: 16px;
      color: var(--color-icon-alternative);
      font-size: 1rem;
    }
  }

  .permission-label-item {
    font-weight: bold;
  }
}
