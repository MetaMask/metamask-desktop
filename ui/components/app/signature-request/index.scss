@import 'signature-request-footer/index';
@import 'signature-request-header/index';
@import 'signature-request-message/index';
@import 'signature-request-data/index';

.signature-request {
  display: flex;
  flex: 1 1 auto;
  flex-direction: column;
  min-width: 0;

  @include screen-sm-min {
    flex: initial;
  }
}

.signature-request-header {
  flex: 1;

  .network-display {
    justify-content: flex-end;
    margin-left: auto;
  }
}

.signature-request-content {
  flex: 1 40%;
  margin-top: 1rem;
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-bottom: 25px;
  min-height: min-content;

  &__title {
    @include H5;

    font-weight: 500;
  }

  &__identicon-container {
    padding: 1rem;
    flex: 1;
    position: relative;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  &__identicon-border {
    height: 75px;
    width: 75px;
    border-radius: 50%;
    border: 1px solid var(--color-background-default);
    position: absolute;
  }

  &__identicon-initial {
    position: absolute;
    font-style: normal;
    font-weight: 500;
    font-size: 60px;
    color: var(--color-background-default);
    z-index: 1;
    text-shadow: var(--shadow-size-xs) var(--color-shadow-default);
  }

  &__info {
    @include H7;

    padding: 0 12px 4px;
  }

  &__info--bolded {
    @include Paragraph;

    font-weight: 500;
  }

  p {
    @include H6;

    color: var(--color-text-muted);
  }
}

