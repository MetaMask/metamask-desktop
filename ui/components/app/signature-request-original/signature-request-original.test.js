import React from 'react';
import configureMockStore from 'redux-mock-store';
import { MESSAGE_TYPE } from '../../../../shared/constants/app';
import mockState from '../../../../test/data/mock-state.json';
import { renderWithProvider } from '../../../../test/lib/render-helpers';
import SignatureRequestOriginal from '.';

const MOCK_SIGN_DATA = JSON.stringify({
  domain: {
    name: 'happydapp.website',
  },
  message: {
    string: 'haay wuurl',
    number: 42,
  },
  primaryType: 'Mail',
  types: {
    EIP712Domain: [
      { name: 'name', type: 'string' },
      { name: 'version', type: 'string' },
      { name: 'chainId', type: 'uint256' },
      { name: 'verifyingContract', type: 'address' },
    ],
    Group: [
      { name: 'name', type: 'string' },
      { name: 'members', type: 'Person[]' },
    ],
    Mail: [
      { name: 'from', type: 'Person' },
      { name: 'to', type: 'Person[]' },
      { name: 'contents', type: 'string' },
    ],
    Person: [
      { name: 'name', type: 'string' },
      { name: 'wallets', type: 'address[]' },
    ],
  },
});

describe('SignatureRequestOriginal', () => {
  const store = configureMockStore()(mockState);

  it('should match snapshot', () => {
    const props = {
      signMessage: jest.fn(),
      cancelMessage: jest.fn(),
      txData: {
        msgParams: {
          from: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
          data: MOCK_SIGN_DATA,
          origin: 'https://happydapp.website/governance?futarchy=true',
        },
        type: MESSAGE_TYPE.ETH_SIGN,
      },
    };

    const { container } = renderWithProvider(
      <SignatureRequestOriginal {...props} />,
      store,
    );

    expect(container).toMatchSnapshot();
  });
});
