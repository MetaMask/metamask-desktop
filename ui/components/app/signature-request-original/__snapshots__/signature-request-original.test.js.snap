// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignatureRequestOriginal should match snapshot 1`] = `
<div>
  <div
    class="request-signature__container"
  >
    <div
      class="request-signature__header"
    >
      <div
        class="request-signature__header-background"
      />
      <div
        class="request-signature__header__text"
      >
        Signature request
      </div>
      <div
        class="request-signature__header__tip-container"
      >
        <div
          class="request-signature__header__tip"
        />
      </div>
    </div>
    <div
      class="request-signature__body"
    >
      <div
        class="request-signature__account-info"
      >
        <div
          class="request-signature__account"
        >
          <div
            class="request-signature__account-text"
          >
            Account:
          </div>
          <div
            class="request-signature__account-item"
          >
            <div
              class="account-list-item undefined"
              data-testid="account-list-item"
            >
              <div
                class="account-list-item__top-row"
              >
                <div
                  class=""
                >
                  <div
                    class="identicon account-list-item__identicon"
                    style="height: 18px; width: 18px; border-radius: 9px;"
                  >
                    <div
                      style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 18px; height: 18px; display: inline-block; background: rgb(250, 58, 0);"
                    >
                      <svg
                        height="18"
                        width="18"
                        x="0"
                        y="0"
                      >
                        <rect
                          fill="#18CDF2"
                          height="18"
                          transform="translate(-0.5897213458840913 -1.8586597890715306) rotate(328.9 9 9)"
                          width="18"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#035E56"
                          height="18"
                          transform="translate(-10.292884711218024 5.9582598028585885) rotate(176.2 9 9)"
                          width="18"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#F26602"
                          height="18"
                          transform="translate(9.375661135250958 -7.990391094185859) rotate(468.9 9 9)"
                          width="18"
                          x="0"
                          y="0"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="account-list-item__account-name"
                >
                  Test Account
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="request-signature__request-icon"
        >
          <div
            class="identicon__image-border"
            style="height: 40px; width: 40px; border-radius: 20px;"
          />
        </div>
        <div
          class="request-signature__balance"
        >
          <div
            class="request-signature__balance-text"
          >
            Balance:
          </div>
          <div
            class="request-signature__balance-value"
          >
            0 ETH
          </div>
        </div>
      </div>
      <div
        class="request-signature__origin-row"
      >
        <div
          class="request-signature__origin-label"
        >
          Origin:
        </div>
        <div
          class="site-origin request-signature__origin"
        >
          <span>
            https://happydapp.website/governance?futarchy=true
          </span>
        </div>
      </div>
      <div
        class="request-signature__notice request-signature__warning"
      >
        Signing this message can be dangerous. This signature could potentially perform any operation on your account's behalf, including granting complete control of your account and all of its assets to the requesting site. Only sign this message if you know what you're doing or completely trust the requesting site.
        <span
          class="request-signature__help-link"
        >
          Learn more
        </span>
      </div>
      <div
        class="request-signature__rows"
      >
        <div
          class="request-signature__row"
        >
          <div
            class="request-signature__row-title"
          >
            Message:
          </div>
          <div
            class="request-signature__row-value"
          >
            {"domain":{"name":"happydapp.website"},"message":{"string":"haay wuurl","number":42},"primaryType":"Mail","types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Group":[{"name":"name","type":"string"},{"name":"members","type":"Person[]"}],"Mail":[{"name":"from","type":"Person"},{"name":"to","type":"Person[]"},{"name":"contents","type":"string"}],"Person":[{"name":"name","type":"string"},{"name":"wallets","type":"address[]"}]}}
          </div>
        </div>
      </div>
    </div>
    <div
      class="request-signature__footer"
    >
      <button
        class="button btn--rounded btn-secondary btn--large request-signature__footer__cancel-button"
        role="button"
        tabindex="0"
      >
        Cancel
      </button>
      <button
        class="button btn--rounded btn-primary btn--large request-signature__footer__sign-button"
        data-testid="request-signature__sign"
        role="button"
        tabindex="0"
      >
        Sign
      </button>
    </div>
  </div>
</div>
`;
